### ISSUE TITLE
Allow using array for constructing binary trees

### ISSUE THREAD

**Issue by czgdp1807** [codezonediitj/pydatastructs#39]
#### Description of the problem

Idea of using arrays for constructing binary trees originated from https://github.com/codezonediitj/pydatastructs/issues/38#issuecomment-562864441 which uses for constructing for binary heaps.
The challenge is to make the change without breaking existing API. One approach can be using default argument like, `elements=[]` and the transformed API would look like, 
```python
def __new__(cls, elements=[], key=None, root_data=None, comp=None,
                is_order_statistic=False):
```
If the above is done then the existing tests for binary_trees.py can be shortened.

**Comment by 23umesh** [comment]
This can be done by passing the array of elements and in `__new__` we can insert every element of passed array in `object.tree` , right?

**Comment by czgdp1807** [comment]
Yes we can do that. A PR would be better for explaining the solution.

**Comment by akshaygoelakki** [comment]
i want to work on this issue..please assign this to me

**Comment by czgdp1807** [comment]
@akshaygoelakki Feel free to start your work on this. Whenever you will create a PR, it will  be linked to this issue and it will show up in assignee section.

**Comment by nikipr1999** [comment]
I want to work on this issue.  So kindly assign this issue to me .

**Comment by Jahnavi-Jonnalagadda** [comment]
@czgdp1807 Could you please provide the link for this code where the changes are to be made?

**Comment by ayushi-get** [comment]
i would like to work on this issue

**Comment by sidhu1012** [comment]
You have it! Start working.

**Comment by sidhu1012** [comment]
You have it! Start working.

**Comment by dollyjain51099** [comment]
I would like to work on this issue, @czgdp1807 kindly assign it to me!

**Comment by czgdp1807** [comment]
@dollyjain51099 We don't assign issues to contributors. Anyone can work on any issue. Link the PR you open with this issue.
